### ISSUE TITLE
predict_proba does not work after freeze()

### ISSUE THREAD

**Issue by angangwa** [huggingface/setfit#281]
Hi, seems like a bug. Unless I'm missing a step?

`trainer.model.predict_proba` raises an error if used after training the differentiable head. 

Code:

```python
model = SetFitModel.from_pretrained(
      "sentence-transformers/paraphrase-mpnet-base-v2",
      use_differentiable_head=True,
      head_params={"out_features": 4},
  )
    
trainer = SetFitTrainer(
    model=model,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
    loss_class=CosineSimilarityLoss,
    batch_size=4,
    num_iterations=20,
    num_epochs=1
)
    
trainer.train()     

trainer.freeze()
trainer.unfreeze(keep_body_frozen=True)

trainer.train(
    num_epochs=10,
    batch_size=4,
    body_learning_rate=1e-5, 
    learning_rate=1e-2, 
    l2_weight=None,
)

print(trainer.evaluate())

trainer.model(list_of_strings) # works
trainer.model.predict_proba(list_of_strings) # does not work

```

```
/opt/conda/lib/python3.7/site-packages/setfit/modeling.py in predict_proba(self, x_test)
    329     def predict_proba(self, x_test: List[str]) -> Union[torch.Tensor, np.ndarray]:
    330         embeddings = self.model_body.encode(x_test, normalize_embeddings=self.normalize_embeddings)
--> 331         return self.model_head.predict_proba(embeddings)
    332 
    333     def __call__(self, inputs):

/opt/conda/lib/python3.7/site-packages/setfit/modeling.py in predict_proba(self, x_test)
    133         self.eval()
    134 
--> 135         return self(x_test)
    136 
    137     def predict(self, x_test: Union[torch.Tensor, "ndarray"]) -> Union[torch.Tensor, "ndarray"]:

/opt/conda/lib/python3.7/site-packages/torch/nn/modules/module.py in _call_impl(self, *input, **kwargs)
   1192         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1193                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1194             return forward_call(*input, **kwargs)
   1195         # Do not call functions when jit is used
   1196         full_backward_hooks, non_full_backward_hooks = [], []

/opt/conda/lib/python3.7/site-packages/setfit/modeling.py in forward(self, features, temperature)
    117 
    118         x = features["sentence_embedding"] if is_features_dict else features
--> 119         logits = self.linear(x)
    120         if self.out_features == 1:  # only has one target
    121             outputs = torch.sigmoid(logits)

/opt/conda/lib/python3.7/site-packages/torch/nn/modules/module.py in _call_impl(self, *input, **kwargs)
   1192         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1193                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1194             return forward_call(*input, **kwargs)
   1195         # Do not call functions when jit is used
   1196         full_backward_hooks, non_full_backward_hooks = [], []

/opt/conda/lib/python3.7/site-packages/torch/nn/modules/linear.py in forward(self, input)
    112 
    113     def forward(self, input: Tensor) -> Tensor:
--> 114         return F.linear(input, self.weight, self.bias)
    115 
    116     def extra_repr(self) -> str:

TypeError: linear(): argument 'input' (position 1) must be Tensor, not numpy.ndarray

```

**Comment by angangwa** [comment]
Directly using the model body and head seem to work:

```python
embeddings = model_frozen.model_body.encode(
    list_of_strings,
    convert_to_tensor=True
)
model_frozen.model_head.predict_proba(embeddings)
```

I was missing the `convert_to_tensor=True`. 

Seems duplicate and related to https://github.com/huggingface/setfit/issues/237. Can close.

**Comment by tomaarsen** [comment]
Hello!

> Seems duplicate and related to https://github.com/huggingface/setfit/issues/237. Can close.

You're exactly right. It's indeed related to the `convert_to_tensor=True`. The issue has been fixed in #207, but I'm afraid I do not have the necessary permissions to publish a release.

Until a new release is pushed, likely some time in early February, you can either use SetFit from the GitHub repository directly. Off the top of my head that involves:
```
pip install git+https://github.com/huggingface/setfit.git
```
or you can use exactly the workaround that you described.

I'll close this for now. 

- Tom Aarsen
